### ISSUE TITLE
Pagination breaks the inventory

### ISSUE THREAD

**Issue by Mr-EmPee** [RyseInventory/RyseInventory#60]
**Describe the bug**
When I set the items that should appear inside my paginated menu the inventory gets bugged and doesn't work as expected

**To Reproduce**
```java

@RequiredArgsConstructor
public class BarrierListGUI implements Bean {

  private final JavaPlugin plugin;
  private final BarrierService barrierService;

  public RyseInventory get() {
    return RyseInventory.builder()
      .title(" ")
      .rows(5)
      .disableUpdateTask()
      .provider(buildProvider())
      .build(plugin);
  }

  private ItemStack getBorderItem() {
    return ItemBuilder.from(Material.BLUE_STAINED_GLASS_PANE)
      .setName(" ")
      .build();
  }

  private void populateBorder(InventoryContents contents) {
    contents.fillBorders(getBorderItem());
    contents.set(3, 1, getBorderItem());
    contents.set(3, 7, getBorderItem());
  }

  private ItemStack getBarrierItem(Barrier barrier) {
    return ItemBuilder.from(Material.BOOK)
      .setName(ChatColor.YELLOW + barrier.getId())
      .setLore("", "", "")
      .build();
  }

  private Consumer<InventoryClickEvent> onBarrierItemClick(Barrier barrier) {
    return event -> {
      //TODO: Open barrier editing menu
      event.getWhoClicked().sendMessage("WIP");
    };
  }

  private List<IntelligentItem> getBarriersItems() {
    return barrierService.findAll().stream()
      .map(b -> IntelligentItem.of(getBarrierItem(b), onBarrierItemClick(b)))
      .toList();
  }

  private IntelligentItem getNextPageButtonItem(Pagination pagination) {
    return IntelligentItem.of(
      ItemBuilder.from(pagination.isLast() ? Material.ENDER_EYE : Material.ENDER_PEARL)
        .setName("&eNext page")
        .build(),
      e -> {
        if(!pagination.isLast()) {
          pagination.inventory().open((Player) e.getWhoClicked(), pagination.next().page());
        }
      }
    );
  }

  private IntelligentItem getPreviousPageButtonItem(Pagination pagination) {
    return IntelligentItem.of(
      ItemBuilder.from(pagination.isFirst() ? Material.ENDER_EYE : Material.ENDER_PEARL)
        .setName("&ePrevious page")
        .build(),
      e -> {
        if(!pagination.isFirst()) {
          pagination.inventory().open((Player) e.getWhoClicked(), pagination.previous().page());
        }
      }
    );
  }

  private InventoryProvider buildProvider() {
    return new InventoryProvider() {
      @Override
      public void init(Player player, InventoryContents contents) {
        populateBorder(contents);

        Pagination pagination = contents.pagination();
        pagination.iterator(SlotIterator.builder()
          .startPosition(1, 1)
          .endPosition(3, 7)
          .blackList(28, 34)
          .build());

        //If I uncomment this line VVV the item inside row 4 col 3 isn't shown
        //pagination.setItems(getBarriersItems());

        contents.set(4, 3, getPreviousPageButtonItem(pagination));
        contents.set(4, 5, getNextPageButtonItem(pagination));
      }
    };
  }
```

**Expected behavior**
The item on row 4 and col 3 to appear

**Comment by Rysefoxx** [comment]
This bug is fixed in version 1.5.8.

**Comment by Mr-EmPee** [comment]
Okay, gonna wait for the next release!

**Comment by Rysefoxx** [comment]
Should be fixed in version v1.6.0.
https://github.com/RyseInventory/RyseInventory/releases/tag/v1.6.0
